service/frontend/workflowHandler.go

```diff
@@ -1028,19 +1031,24 @@ func (wh *WorkflowHandler) PollActivityTaskQueue(ctx context.Context, request *w
 
 	err = backoff.Retry(op, frontendServiceRetryPolicy, common.IsServiceTransientError)
 	if err != nil {
-		errCancel := wh.cancelOutstandingPoll(ctx, err, namespaceID, enumspb.TASK_QUEUE_TYPE_ACTIVITY, request.TaskQueue, pollerID)
-		if errCancel != nil {
-			// For all other errors log an error and return it back to client.
-			ctxTimeout := "not-set"
-			ctxDeadline, ok := ctx.Deadline()
-			if ok {
-				ctxTimeout = ctxDeadline.Sub(callTime).String()
-			}
-			wh.GetLogger().Error("PollActivityTaskQueue failed.",
-				tag.WorkflowTaskQueueName(request.GetTaskQueue().GetName()),
-				tag.Value(ctxTimeout),
-				tag.Error(errCancel))
+		err = wh.cancelOutstandingPoll(ctx, err, namespaceID, enumspb.TASK_QUEUE_TYPE_ACTIVITY, request.TaskQueue, pollerID)
+		// cancelOutstandingPoll returns nil error only if it caller context was cancelled.
+		if err == nil {
+			// It doesn't matter what to return here, client has already gone. But (nil,nil) is invalid gogo return pair.
+			return &workflowservice.PollActivityTaskQueueResponse{}, nil
```

Comment:
why returning any error if poll fails?

Reply:
I refactored code for better readability. We don't want context cancellations (=worker goes down) count against server error metrics as it is not a server error.